File: Backend/Eventora/src/main/java/com/dimitar/eventora/model/Event.java
package com.dimitar.eventora.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import com.dimitar.eventora.model.SeatingLayout;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Event {
    private Long id;
    private String name;
    private String description;
    private LocalDateTime eventDate;
    private Genre genre;
    private BigDecimal ticketPrice;
    private Integer maxTickets;
    private Integer availableTickets;
    private String imageUrl;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean isActive;
    private Long organizerId;
    private Boolean hasSeating;
    private SeatingLayout seatingLayout;
    private Integer seatedCapacity;
    private Integer standingCapacity;

    public boolean canBookTickets(int quantity) {
        return this.isActive && this.availableTickets >= quantity;
    }

    public void bookTickets(int quantity) {
        if (!canBookTickets(quantity)) {
            throw new IllegalStateException("Cannot book tickets for this event");
        }
        this.availableTickets -= quantity;
    }

    public boolean isUpcoming() {
        return this.eventDate.isAfter(LocalDateTime.now());
    }

    public boolean isFull() {
        return this.availableTickets <= 0;
    }
}

File: Frontend/src/types/event/event.ts
import { IGenre } from "./genre"

export interface IEvent {
  id: number
  name: string
  description: string
  eventDate: string
  genre: IGenre
  ticketPrice: number
  maxTickets: number
  availableTickets: number
  standingCapacity: number
  seatedCapacity: number
  hasSeating: boolean
  seatingLayout: "NONE" | "FLOOR" | "FLOOR_BALCONY"
  imageUrl?: string
  isActive: boolean
  organizerId: number
  createdAt: string
  updatedAt: string
}

export interface IEventFormData {
  name: string
  description: string
  eventDate: string
  genre: string
  ticketPrice: number | string
  standingCapacity: number | string
  hasSeating: boolean
  seatingLayout: "NONE" | "FLOOR" | "FLOOR_BALCONY"
  imageUrl: string
}

export interface ITicket {
  id: string
  event: IEvent
  qrCode: string
  status: "valid" | "used" | "expired"
  purchaseDate: string
  usedAt?: string
  purchasePrice: number
  number: string
}

// Export non-prefixed versions for backward compatibility
export type Event = IEvent
export type EventFormData = IEventFormData
export type Ticket = ITicket
